profiles/permissions.py

```diff
@@ -20,3 +23,38 @@ def has_object_permission(self, request, view, profile):
             return True
 
         return profile.user == request.user
+
+
+class CanEditIfPrivate(BasePermission):
+    """
+    Dont allow users to view private profiles.
+    Also dont allow anonymous users or non verified mm user to view profiles with
+    account_privacy is set to public public_to_mm.
+    """
+
+    def has_permission(self, request, view):
+        """
+        Only owner of a profile can view and edit their profile.
+        Verified micromaster users can view other profiles to limited view only if
+        account_privacy of profile is set to public public_to_mm
+        """
+        filter_kwargs = {'user__social_auth__uid': view.kwargs['user']}
+        profile = get_object_or_404(view.queryset, **filter_kwargs)
+
+        if request.user == profile.user:
+            return True
+
+        # private profiles
+        if profile.account_privacy == Profile.PRIVATE:
+            raise Http404
+        elif profile.account_privacy == Profile.PUBLIC_TO_MM:
+            # anonymous user accessing profiles.
+            if request.user.is_anonymous():
+                raise Http404
```

Comment:
Sorry I misunderstood. 

Still, an anonymous user hasn't agreed to the terms of service, or invested in the community. They shouldn't get to know who is a member. 

I'm trying to think of a use case where an anonymous user needs minimal access to a profile -- perhaps to see avatars in the forum -- but that won't be available to an anonymous user either.